### PR TITLE
Some tweaks to the editing API docs

### PR DESCRIPTION
**Pull request by kaedroho**

### CODE REVIEW

**Comment by kaedroho** [comment]
Please merge this before making any more changes to the editing api docs get made. I'm not willing to rebase this if a conflict arises.

File: docs/reference/pages/editing_api.rst [hunk 1]
```diff
@@ -18,232 +18,248 @@ Defining Panels
 
 A "panel" is the basic editing block in Wagtail. Wagtail will automatically pick the appropriate editing widget for most Django field types; implementors just need to add a panel for each field they want to show in the Wagtail page editor, in the order they want them to appear.
 
-There are four basic types of panels:
+Wagtail provides a tabbed interface to help organize panels. Three such tabs are provided:
```
Comment (tomdyson) [inline docs/reference/pages/editing_api.rst]: organize->organise - let's standardise on 'ise'.

File: docs/reference/pages/editing_api.rst [hunk 2]
```diff
@@ -18,232 +18,248 @@ Defining Panels
...
+
+* ``content_panels`` is the main tab, used for the bulk of your model's fields.
+* ``promote_panels`` is suggested for organizing fields regarding the promotion of the page around the site and the Internet. For example, a field to dictate whether the page should show in site-wide menus, descriptive text that should appear in site search results, SEO-friendly titles, OpenGraph meta tag content and other machine-readable information.
```
Comment (tomdyson) [inline docs/reference/pages/editing_api.rst]: izing -> ising

File: docs/reference/pages/editing_api.rst [hunk 3]
```diff
@@ -18,232 +18,248 @@ Defining Panels
...
+      ImageChooserPanel('splash_image'),
+      DocumentChooserPanel('free_download'),
+      PageChooserPanel('related_page'),
+    ]
+
+    promote_panels = [
+      MultiFieldPanel(Page.promote_panels, "Common page configuration"),
+    ]
+
+After the ``Page``-derived class definition, just add lists of panel definitions to order and organize the Wagtail page editing interface for your model.
```
Comment (tomdyson) [inline docs/reference/pages/editing_api.rst]: as above

File: docs/reference/pages/editing_api.rst [hunk 4]
```diff
@@ -18,232 +18,248 @@ Defining Panels
...
-
-  ExamplePage.promote_panels = [
-    MultiFieldPanel(COMMON_PANELS, "Common page configuration"),
-  ]
+.. class:: InlinePanel(relation_name, panels=None, classname=None, label='', help_text='')
 
-After the ``Page``-derived class definition, just add lists of panel definitions to order and organize the Wagtail page editing interface for your model.
+    This panel allows for the creation of a "cluster" of related objects over a join to a separate model, such as a list of related links or slides to an image carousel.
 
+    This is a very powerful, but tricky feature which will take some space to cover, so we'll skip over it for now. For a full explanation on the usage of ``InlinePanel``, see :ref:`inline_panels`.
```
Comment (tomdyson) [inline docs/reference/pages/editing_api.rst]: -> "This is a powerful but tricky feature"

File: docs/reference/pages/editing_api.rst [hunk 5]
```diff
@@ -18,232 +18,248 @@ Defining Panels
...
 
-Here are some Wagtail-specific types that you might include as fields in your models.
+    This panel creates a columnar layout in the editing interface, where each of the child Panels appears alongside each other rather than below.
 
+    Use of FieldRowPanel particularly helps reduce the "snow-blindness" effect of seeing so many fields on the page, for complex models. It also improves the perceived association between fields of a similar nature. For example if you created a model representing an "Event" which had a starting date and ending date, it may be intuitive to find the start and end date on the same "row".
 
-Rich Text (HTML)
-----------------
+    FieldRowPanel should be used in combination with ``col*`` classnames added to each of the child Panels of the FieldRowPanel. The Wagtail editing interface is layed out using a grid system, in which the maximum width of the editor is 12 columns wide. Classes ``col1``-``col12`` can be applied to each child of a FieldRowPanel. The class ``col3`` will ensure that field appears 3 columns wide or a quarter the width. ``col4`` would cause the field to be 4 columns wide, or a third the width.
```
Comment (tomdyson) [inline docs/reference/pages/editing_api.rst]: layed out -> laid out. 12 columns wide -> 12 columns.

File: docs/reference/pages/editing_api.rst [hunk 6]
```diff
@@ -18,232 +18,248 @@ Defining Panels
...
+    ``PageChooserPanel`` takes two arguments: a field name and an optional page type. Specifying a page type (in the form of an ``"appname.modelname"`` string) will filter the chooser to display only pages of that type.
 
-.. _Django model field reference (on_delete): https://docs.djangoproject.com/en/dev/ref/models/fields/#django.db.models.ForeignKey.on_delete
+ImageChooserPanel
+-----------------
 
-Displaying ``Image`` objects in a template requires the use of a template tag. See :ref:`image_tag`.
+.. class:: ImageChooserPanel(field_name)
 
+    One of the features of Wagtail is a unified image library, which you can access in your models through the ``Image`` model and the ``ImageChooserPanel`` chooser. Here's how:
```
Comment (tomdyson) [inline docs/reference/pages/editing_api.rst]: -> "Wagtail includes a unified image library, which you can..."

File: docs/reference/pages/editing_api.rst [hunk 7]
```diff
@@ -18,232 +18,248 @@ Defining Panels
...
+              related_name='+'
+          )
 
-As with images, Wagtail documents should also have the appropriate extra parameters to prevent cascade deletions across a ForeignKey relationship. ``DocumentChooserPanel`` takes only one argument: the name of the field.
+          content_panels = Page.content_panels + [
+              ImageChooserPanel('cover'),
+          ]
 
-Documents can be used directly in templates without tags or filters. Its properties are:
+    Django's default behavior is to "cascade" deletions through a ForeignKey relationship, which is probably not what you want happening. This is why the ``null``, ``blank``, and ``on_delete`` parameters should be set to allow for an empty field. (See `Django model field reference (on_delete)`_ ). ``ImageChooserPanel`` takes only one argument: the name of the field.
```
Comment (tomdyson) [inline docs/reference/pages/editing_api.rst]: "which is probably not what you want happening" -> "which may not be what you want"

File: docs/reference/pages/editing_api.rst [hunk 8]
```diff
@@ -18,232 +18,248 @@ Defining Panels
...
+              SnippetChooserPanel('advert', Advert),
+          ]
 
-See :ref:`snippets` for more information.
+    See :ref:`snippets` for more information.
+
+Built-in Fields and Choosers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Django's field types are automatically recognized and provided with an appropriate widget for input. Just define that field the normal Django way and pass the field name into ``FieldPanel()`` when defining your panels. Wagtail will take care of the rest.
```
Comment (tomdyson) [inline docs/reference/pages/editing_api.rst]: recognised

File: docs/reference/pages/editing_api.rst (hunk 1 above)
Comment (kaedroho) [inline docs/reference/pages/editing_api.rst]: I'll run the spellchecker over this which should catch all these

**Comment by tomdyson** [comment]
I'm happy for this to be merged with my changes. Thanks, @kaedroho!

**Pull request merged**